src/vaadin-menu-bar-interactions-mixin.html

```diff
@@ -0,0 +1,342 @@
+<!--
+@license
+Copyright (c) 2019 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+
+<script>
+  /**
+   * @namespace Vaadin
+   */
+  window.Vaadin = window.Vaadin || {};
+
+  /**
+   * @namespace Vaadin.MenuBar
+   */
+  window.Vaadin.MenuBar = window.Vaadin.MenuBar || {};
+
+  /**
+   * @polymerMixin
+   */
+  Vaadin.MenuBar.InteractionsMixin = superClass => class InteractionsMixin extends superClass {
+
+    static get properties() {
+      return {
+        /**
+         * If true, the nested menu will open on hover (mouseover) instead of click.
+         */
+        openOnHover: {
+          type: Boolean
+        }
+      };
+    }
+
+    constructor() {
+      super();
+      this.__boundOutsideClickListener = this.__outsideClickListener.bind(this);
+      this.__boundOnContextMenuKeydown = this.__onContextMenuKeydown.bind(this);
+    }
+
+    static get observers() {
+      return [
+        '_itemsChanged(items, items.splices)'
+      ];
+    }
+
+    ready() {
+      super.ready();
+
+      this.addEventListener('keydown', e => this._onKeydown(e));
+      this.addEventListener('focusin', e => this._onFocusin(e));
+
+      this._subMenu.addEventListener('item-selected', this.__onItemSelected.bind(this));
+      this._subMenu.addEventListener('close-all-menus', this.__onEscapeClose.bind(this));
```

Comment:
The menu should also close by Esc key when the focus is on a button.
To reproduce:
1. Open a sub-menu
2. Hit the Left arrow key to open the previous sub-menu while keeping the focus on the button instead of an item
3. Hit the Esc key and notice that the menu is not closed

In the w3-example the menu closes in a similar case.

Reply:
Good catch. Will fix.

Reply:
Fixed.